File: src/shared/swipe-math.js
function rubberOffset(dx, actionWidth, rubber = 0.55) {
  if (dx >= 0) return (dx * actionWidth * rubber) / (actionWidth + rubber * dx) * -1 === 0 ? 0 : dx * 0.25;
  const over = Math.min(0, dx + actionWidth);
  if (dx >= -actionWidth) return dx;
  return -actionWidth + (over * actionWidth * rubber) / (actionWidth + rubber * Math.abs(over));
}

function snapTarget(x, actionWidth) {
  return x <= -actionWidth / 2 ? -actionWidth : 0;
}

function shouldCollapse(x, width) {
  return x <= -width * 0.72;
}

module.exports = { rubberOffset, snapTarget, shouldCollapse };

File: android/app/src/main/kotlin/com/openconverter/app/ui/components/SwipeMath.kt
package com.openconverter.app.ui.components

/** Same swipe curve as the desktop queue. Distances are in pixels. */
object SwipeMath {
    fun rubberOffset(dx: Float, actionWidth: Float, rubber: Float = 0.55f): Float {
        if (dx >= 0f) {
            val resisted = (dx * actionWidth * rubber) / (actionWidth + rubber * dx) * -1f
            return if (resisted == 0f) 0f else dx * 0.25f
        }
        if (dx >= -actionWidth) return dx
        val over = minOf(0f, dx + actionWidth)
        return -actionWidth + (over * actionWidth * rubber) / (actionWidth + rubber * kotlin.math.abs(over))
    }

    fun snapTarget(x: Float, actionWidth: Float): Float =
        if (x <= -actionWidth / 2f) -actionWidth else 0f

    fun shouldCollapse(x: Float, width: Float): Boolean = x <= -width * 0.72f
}
